### ISSUE TITLE
PostHog CRM

### ISSUE THREAD

**Issue by jamesefhawkins** [PostHog/posthog#25708]
## Is your feature request related to a problem?

When you're building a new product, most companies will need to sell it.

Early on, startups track deals in their heads, in a spreadsheet, in apple notes, or in airtable.

Let's provide them with something a little better in the tool they're already using - PostHog.

## Describe the solution you'd like

The major features of our CDP should be something like:

* Concept of "a deal" attached to a company (perhaps beef up our "groups" product a bit at the same time)
* Concept of a pipeline with customizable steps - where deals can progress from prospecting to closed/won
* Non MVP but impotant: Email and meeting tracking - so you can easily see the communication history with prospects

We can win this long term by offering a CDP with all the other data inherently in it that we track (recordings/replays/messages/usage/errors) for a better view of customer behavior without integration needed or bad UX for coping with integrations.

## Describe alternatives you've considered

Screwing around with more saas / more integrations

## Additional context

CRMs are very popular integrations, so heavy overlap with our customers.

It's tempting to build a PLG based CRM, but that apparently is a tiny market. Just build a regular CRM first.

### Debug info

```shell
- [ ] PostHog Cloud, Debug information: [please copy/paste from https://us.posthog.com/settings/project-details#variables]
- [ ] PostHog Hobby self-hosted with `docker compose`, version/commit: [please provide]
- [ ] PostHog self-hosted with Kubernetes (deprecated, see [`Sunsetting Kubernetes support`](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: [please provide]
```

**Comment by ScottWilliamAnderson** [comment]
Some of the lessons I've learned over the past few years on what information to prioritise, and how to store it:

- There's not much you can do with limited information. The barrier being lower than ever for the CRM to ingest information is crucial, passive/automated collection is ideal. This even includes information outside our own software, like emails, meeting minutes, exhanged documents, and market indicators.
- Watching a film is not the same as seeing many pictures. There is key insight found in higher levels than just text, like the time between events, reliability of sources for dubious or unconfirmed information, habits, strategy success metrics. 
- A model of a delimited environment which you can change the inputs and even the foundational truths of (no model is perfect but some are useful), is simpler than it is to parse an ever appending stream of collected information. This is the difference between a snapshot that provides context and a continuous feed that lacks focus and cohesion.
- Wherever possible, log changes with associated reasoning to the model.
- You can organise the customer and their environment with Design Thinking. Empathise, Define, Ideate, Prototype, Test. The CRM takes the role of the first two steps. The Project Management Systems will pick up the rest. The user(s) will take responsibility for moving along the pipeline (even asynchronously), allow them to dictate the pace.
- What moves through the pipeline is information, under many shapes. If you can tag and label information, you can describe relationships between all forms of it. The rules & patterns they follow and how they interact, teach you the model of the environment.

Or, just pick your favourite CRM and mirror what they do, that's usually a safe bet!

Hope this helps spark ideas,
Best regards

**Comment by lricoy** [comment]
Hey @jamesefhawkins , this would be a great feature to have, but I would like to use the opportunity to propose another approach. It may be too much for what is being described here, but I think it could be a great addition to PostHog.

tl;dr:

What about a simpler Zapier/n8n/Make/Airflow right inside PostHog?  This would allow for CRM functionality and automation of many other processes, all within a single platform.

Longer braindumped text:

From what is being described, I think this could be bigger than just a B2B CRM and even apply to a PLG CRM as well. PostHog's CDP and warehouse already provide a great foundation for data ingestion and enrichment. When I was using it, exporting data easily, unlike some competitors who lock it behind enterprise paywalls, was a huge selling point because we still needed to keep our other tools working.

But what if we could automate entire sales processes, or any other process, leveraging this enriched data?

From an engineering perspective, imagine a custom automation flow pipeline integrated with experiments, cohorts, the CDP, and webhooks. This would mirror what many users currently achieve with external tools like Zapier or n8n. 

The current limitation of the plugin server is its time-based constraint of ingestion-time only behaviour (from what I can tell). By unlocking the ability to perform actions like "if the user has not been active in the last 30 days, send them an email" right inside of Posthog, the possibilities for automation within PostHog become immense. (Looking at how the cohort and some CRM plugins are used, that one looks popular I was even using it myself)

Like you said, this is probably happening already looking at the popularity. Even the [Plugin Server docs](https://posthog.com/docs/how-posthog-works/ingestion-pipeline#plugin-server) highlight the need for external tools like Hubspot and Customer.io to handle tasks like sending emails and creating tasks. 

Another datapoint that may be relad is: "Messaging" is the second most voted feature on the roadmap, and webhooks are in trial.  With a built-in automation tool, alognside those features and the cohorts, surveys and experiments, that is a full-fledge CRM for many companies, specially the ones that look like Posthog's ICP.

Would love to contribute to that if you think it's a good idea.

Cheers!

**Comment by jamesefhawkins** [comment]
> What about a simpler Zapier/n8n/Make/Airflow right inside PostHog? This would allow for CRM functionality and automation of many other processes, all within a single platform.

I could definitely see this as why it's just cooler having all your business logic in us (for example, our work on Hog, will lead to this stuff). We think it's somewhat "second order" - ie if people are using the "basic" products underneath, then that puts us in the spot to have _all_ the data so that we can build these workflows on top. If you stay tuned with us, I am certain we'll wind up with what you're asking for here.

**Comment by yanchuk** [comment]
@jamesefhawkins is that something aligned with your CRM vision? A few slides: https://pitch.com/v/posthog-crm-vc65yf I prepared recently. I agree with the basics, but it could be bigger, thou MVP is simplier

**Comment by cove-cyril** [comment]
Hey Posthog Team,
Are you guys still working on this?
All the general CRM options are pretty crappy. Would love to have all my user data in one place.

**Comment by anassabbane** [comment]
In house CRM solution is the missing piece. 
Hope this is still being worked on.

**Comment by tonyzorin** [comment]
Hey James,
Any timeline for this?
IMHO a CRM for SaaS businesses has a great potential (and we need one).
